source/logrepl/internal/subscription.go

```diff
@@ -61,18 +61,27 @@ type Handler func(context.Context, pglogrepl.Message, pglogrepl.LSN) (pglogrepl.
 // CreateSubscription initializes the logical replication subscriber by creating the replication slot.
 func CreateSubscription(
 	ctx context.Context,
-	conn *pgconn.PgConn,
+	pool *pgxpool.Pool,
 	slotName,
 	publication string,
 	tables []string,
 	startLSN pglogrepl.LSN,
 	h Handler,
 ) (*Subscription, error) {
-	var slotLSN pglogrepl.LSN
+	// Request a replication connection
+	conn, err := pool.Acquire(context.WithValue(ctx, cpool.WithReplCtxKey, true))
+	if err != nil {
+		return nil, fmt.Errorf("could not establish replication connection: %w", err)
+	}
+	defer func() { // release connection on error
+		if err != nil {
```

Comment:
To make this work reliably, we need to make the returned error a named variable `err`.

Reply:
Ooh, I had it and it dropped off in the refactor, thanks for catching that.

Reply:
Fixed in 6708b6d